src/main/java/example/security/ElideSpringSecurityContextConfig.java

```diff
@@ -0,0 +1,53 @@
+package example.security;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.web.authentication.SimpleUrlAuthenticationFailureHandler;
+
+/**
+ * Example setup for overriding the principal in security context to support elide-async.
+ * Remove this class if not using Async
+ */
+@EnableWebSecurity
+public class ElideSpringSecurityContextConfig extends WebSecurityConfigurerAdapter {
```

Comment:
I don't think we want to provide a default security context here - at least one with authentication.  We want the ergonomics to be really simple to use.  Setting up authentication for a demo is one more thing to frustrate someone who just wants to try things out.